README.md

```diff
@@ -1,13 +1,408 @@
+![Image of Kesiev](http://i.imgur.com/GnCHqIm.png?1) 
 
+Currently, processing collections in C++ is not only far from perfect, but is syntax-wise far behind many current functional programming paradigms.
 
+Languages are used as tools to communicate, and incidentally, the structure and limitations of a C++ determine the way we think. Learning how to think and within the context of a language provides one the ability to be a proper user, well at the same time 
+
+> In 24 hours you might be able to learn some of the syntax of C++ (if you already know another language), but you couldn't learn much about how to use the language. In short, if you were, say, a Basic programmer, you could learn to write programs in the style of Basic using C++ syntax, but you couldn't learn what C++ is actually good (and bad) for. So what's the point? Alan Perlis once said: "A language that doesn't affect the way you think about programming, is not worth knowing" -- [Peter Norvig][1]
+
+[1]:http://www.norvig.com/21-days.html
+
+This is the principle of linguistic relativity also known as the [Sapir-Whorf hypothesis][2] shows that the constraints of a language are what most often limit our thought processing. This idea, when applied to programming languages shows how Lisp programmers use macros for everything, Python programmers even in Ruby have code that looks like python.  
+
+[2]:http://en.wikipedia.org/wiki/Linguistic_relativity
+
+One of the key components to developing C++ is learnability or more importantly, how easy is it for users to accomplish basic tasks. 
+
+C++ Collections is a C++ 11 library that provides lazy evaluation and functional transformations of the data to lighten the burden of using functional operations such as map, range, and filter for the user. To use C++ Streams, download them from Github and simply *#include collections.h* in the header of your file after dragging it in.
+
+----
+## What is C++ Collections?
+C++ Collections provides abstractions on a set of ordered data. All examples below assume that we are `using namespace cpp_collections`.
+The Class collections is used to support functional style transformations as demonstrated above. Assume you would like to create a list of numbers from 1 to 100, sum them, and then store the value in an int. Functional operators allow these operations to happen in nearly one line of code.
+
+```cpp
+int sum = range(1,101).reduceLeft([](int x, int y) {return x + y;});
+```
+
+The syntax you will note is similar to both Haskell and Scala.
+
+```haskell
+sum = foldl1 (+) [1..100]
+```
+
+```scala
+val sum = Range(1, 100).reduce((a,b) => a+b)
+```
+
+The main abstraction introduced by this library is Collection. The capitalization om the word 'Collection' is important to distinguish it from collection. The Collection is different from standard collections in C++ in that:
+
+1. It is functional in nature. All operations on a Collection produce some form of result by modifying the source in memory. 
```

Comment:
Functional means that it isn't modifying the source in place. It's returning a modified copy.
The point is that once you declare X, you can't modify X without reassigning it.